[Evo] Ref #5190 - send email notification to the access requestor

nabdali opened the pull request:
- Add starter mail dependencies
- Create sample email service
- Send email notification to requestor
- Add sample email configuration in config file

nabdali committed:
[Evo] Ref #5190 - send email notification to the access requestor

- Add starter mail dependencies
- Create sample email service
- Send email notification to requestor
- Add sample email configuration in config file

nabdali commented on src/main/kotlin/org/ozwillo/dcimporter/web/DataAccessRequestHandler.kt:
```diff
@@ -14,10 +15,12 @@ import org.springframework.web.reactive.function.server.ServerResponse
 class DataAccessRequestHandler(
-    private val dataAccessRequestRepository: DataAccessRequestRepository
+    private val dataAccessRequestRepository: DataAccessRequestRepository,
+    private val emailService: EmailService
```
I think is beter to use this service in a data access request service when there's going to be one

bobeal replied:
I don't understand what you meant here

nabdali replied:
Sorry for the typo, I meant that we need to create a service for the data access request

bobeal commented on src/main/kotlin/org/ozwillo/dcimporter/web/DataAccessRequestHandler.kt:
```diff
@@ -79,6 +82,16 @@ class DataAccessRequestHandler(
                             )
                         ).subscribe()
+
+                        try {
```
just add a TODO (or file an issue) for a later decoupling of the mail sending feature (but ok for now)

bobeal commented on src/main/kotlin/org/ozwillo/dcimporter/service/EmailService.kt:
```diff
@@ -0,0 +1,19 @@
+        message.setTo(to)
+        message.setSubject(subject)
+        message.setText(text)
+        emailSender!!.send(message)
```
can't it be cleaner (ie no !! enforcing) ?

nabdali committed:
[Evo] Fix #5190 - Minor refactoring

- Minor refactoring suggested in PR (#18)
- Add TODO comment for later improvement
